### PR TITLE
✨ Introduce new naming convention for artifacts

### PR DESCRIPTION
**Pull request by mauromorales**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates to #1109

## Results

Running `earthly +all` produces the following artifacts:

```
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-grype.json
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-grype.sarif
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-initrd
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty.ipxe
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-ipxe.iso.ipxe
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-ipxe-usb.img.ipxe
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty.iso
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty.iso.sha256
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-kernel
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-sbom.spdx.json
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-sbom.syft.json
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty.squashfs
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-trivy.html
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-trivy.json
kairos-core-opensuse-leap-amd64-generic-v2.3.0-11-gc84e074-dirty-trivy.sarif
```

`earthly -P +all-arm --FLAVOR=ubuntu-arm-rpi --COMPRESS_IMG=false`

```
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-grype.json
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-grype.sarif
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty.img
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty.img.sha256
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-sbom.spdx.json
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-sbom.syft.json
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-trivy.html
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-trivy.json
kairos-core-ubuntu-arm64-rpi64-v2.3.0-21-gc683a90-dirty-trivy.sarif
```

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -573,36 +595,44 @@ iso-remote:
...
+    END
 
-   SAVE ARTIFACT /build/$ISO_NAME.squashfs squashfs AS LOCAL build/$ISO_NAME.squashfs
-   SAVE ARTIFACT /build/$ISO_NAME-kernel kernel AS LOCAL build/$ISO_NAME-kernel
-   SAVE ARTIFACT /build/$ISO_NAME-initrd initrd AS LOCAL build/$ISO_NAME-initrd
-   SAVE ARTIFACT /build/$ISO_NAME.ipxe ipxe AS LOCAL build/$ISO_NAME.ipxe
+    SAVE ARTIFACT /build/$ISO_NAME.squashfs squashfs AS LOCAL build/$ISO_NAME.squashfs
+    # TODO: why do the kernel and initrd files don't have an extension?
+    SAVE ARTIFACT /build/$ISO_NAME-kernel kernel AS LOCAL build/$ISO_NAME-kernel
+    SAVE ARTIFACT /build/$ISO_NAME-initrd initrd AS LOCAL build/$ISO_NAME-initrd
```
Comment (mauromorales) [inline Earthfile:619]: why do the kernel and initrd files don't have an extension?

File: Earthfile [hunk 2]
```diff
@@ -573,36 +595,44 @@ iso-remote:
...
+    # TODO: what is the different between this one and the ipxe.iso.ipxe one?
+    SAVE ARTIFACT /build/$ISO_NAME.ipxe ipxe AS LOCAL build/$ISO_NAME.ipxe
```
Comment (mauromorales) [inline Earthfile:621]: what is the different between this one and the ipxe.iso.ipxe one?

File: Earthfile (hunk 1 above)
Comment (Itxaka) [reply Earthfile:619]: what extension could they have? sometimes they migth have img or gz or img.gz so I guess its difficult to infer wha thte correct one should be.

also does it matter?

File: Earthfile [hunk 3]
```diff
@@ -66,16 +69,15 @@ all-arm:
   ARG SECURITY_SCANS=true
   BUILD --platform=linux/arm64 +base-image --MODEL=rpi64
   IF [ "$SECURITY_SCANS" = "true" ]
-      BUILD --platform=linux/arm64 +image-sbom --MODEL=rpi64
-      BUILD --platform=linux/arm64 +trivy-scan --MODEL=rpi64
-      BUILD --platform=linux/arm64 +grype-scan --MODEL=rpi64
+      BUILD --platform=linux/arm64 +image-sbom --MODEL=rpi64 --ARCH=arm64
```
Comment (Itxaka) [inline Earthfile:72]: could we infer the arch directly via what earthly sets in env vars that way we dont have to "manually" pass them and will work everywhere? 

`TARGETARCH` can be `arm, amd64, arm64`

**Review by Itxaka — APPROVED** [review]
looking good to me!

File: Earthfile (hunk 1 above)
Comment (mauromorales) [reply Earthfile:619]: No it doesn't I will update the docs then

File: Earthfile (hunk 3 above)
Comment (mauromorales) [reply Earthfile:72]: Good call

**Pull request merged**